api_fhir_r4/converters/communicationConverter.py

```diff
@@ -181,11 +181,10 @@ def build_imis_asessment(cls, imis_feedback, fhir_content_string):
 
     @classmethod
     def _convert_bool_value(cls, fhir_content_string):
-        if fhir_content_string == "yes":
-            return True
-        if fhir_content_string == "no":
-            return False
-        return None
+        if fhir_content_string not in [ "yes", "no"]:
+            return None
+        
+        return fhir_content_string == "yes"
```

Comment:
Performance-wise, I think that the comparison is better. Also, I would raise an exception if the value is not "yes" or "no".

```suggestion
    @staticmethod
    def _convert_bool_value(fhir_content_code, fhir_content_string):
         if fhir_content_string == "yes":
            return True
        if fhir_content_string == "no":
            return False
        raise ValidationError(f"Value for '{fhir_content_code}' must be either 'yes' or 'no'")
```

Reply:
With this change, it would change the methods calling to an assignation only, without any condition.

Reply:
> Performance-wise, I think that the comparison is better. 

I prefer readability over tiny performance improvement, especially when the performance argument is not certain.

it's true the `not in` will require a full check of the list, and that you might get some improvement.

```python
def test_with_comparison(content):
  if content == "yes": return True
  if content == "no": return False
  return None
 
def test_with_membership(content):
  if content not in ["yes", "no"]: return None
  return content == "yes"

def measure_time_of_test_comparison():
  MILLION_NUMBERS = list(range(1_000_000))
  start = time.time()
  for i in MILLION_NUMBERS: test_with_comparison("yes")
  stop = time.time()
  print(f'compare yes: {stop - start}')
  start = time.time()
  for i in MILLION_NUMBERS: test_with_comparison("no")
  stop = time.time()
  print(f'compare no: {stop - start}')
  start = time.time()
  for i in MILLION_NUMBERS: test_with_membership("yes")
  stop = time.time()
  print(f'membership yes: {stop - start}')
  start = time.time()
  for i in MILLION_NUMBERS: test_with_membership("no")
  stop = time.time()
  print(f'membership no: {stop - start}')
```

gives me the following:

```
compare yes: 0.06908130645751953
compare no: 0.08493781089782715
membership yes: 0.07633566856384277
membership no: 0.08259201049804688
```

so you're right.

nevertheless, I'm going to push another suggestion where I validate independently the user input data (which is a good practice to follow), then I convert it if the data is valid, so we don't mix both concerns.